string_utils/fms_string_utils.F90

```diff
@@ -248,30 +240,52 @@ subroutine fms_f2c_string (dest, str_in)
     enddo
 end subroutine fms_f2c_string
 
-  !> @brief Converts a Boolean value to a string
-  !> @return The Boolean value as a string
-  function string_from_logical(v)
-    logical, intent(in) :: v !< Boolean value to be converted to a string
-    character(:), allocatable :: string_from_logical
-
-    if (v) then
-      string_from_logical = "True"
-    else
-      string_from_logical = "False"
-    endif
+  !> @brief Converts a number or a Boolean value to a string
+  !> @return The argument as a string
+  function string(v, fmt)
+    class(*), intent(in) :: v ! Value to be converted to a string
+    character(*), intent(in), optional :: fmt !< Optional format string for a real argument
+    character(:), allocatable :: string
```

Comment:
By making it `allocatable`, leading and trailing whitespace can be stripped so that the user doesn't need to do `trim(string(x))`. For me at least, this outweighs the benefit of making the function `elemental` (although I'm biased by the fact that I've only used `string()` for generating unit test log messages).